File: src/main/java/engineer/thomas_werner/euler/math/PrimeSequence.java
package engineer.thomas_werner.euler.math;

public class PrimeSequence {

    private long lastNum = 1;

    public long nextValue() {
        while(!isPrime(++lastNum));
        return lastNum;
    }

    public static boolean isPrime(final long number) {
        if(number < 2)
            return false;

        if (0 == number % 2)
            return number == 2;

        for(long l=3; l*l <= number; l+=2)
            if(0 == number % l)
                return false;

        return true;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/sequences/PrimeSequence.kt
package de.huddeldaddel.euler.sequences

class PrimeSequence {

    private var lastNum: Long = 0

    constructor() {
        lastNum = 1
    }

    constructor(initialValue: Long) {
        lastNum = initialValue
    }

    fun nextValue(): Long {
        while (!isPrime(++lastNum));
        return lastNum
    }

    companion object {

        fun isPrime(number: Long): Boolean {
            if (number < 2)
                return false

            if (0L == number % 2)
                return number == 2L

            var l: Long = 3
            while (l * l <= number) {
                if (0L == number % l)
                    return false
                l += 2
            }

            return true
        }
    }

}
